apps/api-extractor/src/generators/ApiModelGenerator.ts

```diff
@@ -817,13 +821,17 @@ export class ApiModelGenerator {
     let apiProperty: ApiProperty | undefined = parentApiItem.tryGetMemberByKey(containerKey) as ApiProperty;
 
     if (apiProperty === undefined) {
-      const propertyDeclaration: ts.PropertyDeclaration =
-        astDeclaration.declaration as ts.PropertyDeclaration;
-
       const nodesToCapture: IExcerptBuilderNodeToCapture[] = [];
 
       const propertyTypeTokenRange: IExcerptTokenRange = ExcerptBuilder.createEmptyTokenRange();
-      nodesToCapture.push({ node: propertyDeclaration.type, tokenRange: propertyTypeTokenRange });
+
+      // If the property declaration's type is `undefined`, then we're processing a setter with no corresponding
+      // getter. Use the parameter type instead (note that TSC will enforce that the setter has exactly one
+      // parameter).
```

Comment:
> note that TSC will enforce that the setter has exactly one parameter

The compiler might enforce it, but we should validate that assumption here.
For example, what if the .d.ts file was manually authored, or what if some new syntax
variant is introduced in a future?

Reply:
Some [experimentation shows](https://www.typescriptlang.org/play?strict=false&noImplicitAny=false&strictNullChecks=false&strictFunctionTypes=false&strictPropertyInitialization=false&strictBindCallApply=false&noImplicitThis=false&noImplicitReturns=false&alwaysStrict=false#code/MYGwhgzhAEAaBQBve1XQOYFMAu0D2A1mAJ4AUAlNMmjdAE44CudAdgNwpoC+nqEO+ImQBulRD17R+uFnmwB5IRSo8eQA) that it's a compile error (TS 1049) if the setter does not have exactly one parameter.  However the compiler happily accepts setters/getters with missing type annotations:

```ts
class X
{
    get okay() {
        return;
    }
    set okay(v) {}

    set notOkay() {} // error TS 20149
}
```

Reply:
...but looking more closely at the code, the `IExcerptBuilderNodeToCapture.node` allows `undefined`.  So your code is functionally correct, we just should eliminate the `!`.

Reply:
I pushed a small fix